performance.Rmd

```diff
@@ -0,0 +1,227 @@
+---
+title: "Optimize `polars` performance"
+output: github_document
+---
+
+
+As highlighted by the [DuckDB benchmarks](https://duckdblabs.github.io/db-benchmark/),
+`polars` is very efficient to deal with large datasets. Let's take a very basic
+example dataset of 10M rows and compute the mean of every column per group:
+
+```{r}
+suppressPackageStartupMessages({
+  library(dplyr)
+  library(polars)
+  library(data.table)
+})
+
+N = 1e7
+df = data.frame(matrix(runif(25 * N), nrow = N))
+df$letters = sample(letters, N, replace = TRUE)
+
+df_dt = setDT(df)
```

Comment:
Out of principle, this should probably be `data.table(df)` to avoid changing `df` itself, and the benchmark would probably be fairer without `setkey()` since the other methods do not benefit from this pre-processing.